lib/promscrape/discovery/ec2/instance.go

```diff
@@ -4,12 +4,19 @@ import (
 	"encoding/xml"
 	"fmt"
 	"strings"
+	"time"
 
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
 )
 
 // getInstancesLabels returns labels for ec2 instances obtained from the given cfg
 func getInstancesLabels(cfg *apiConfig) ([]map[string]string, error) {
+	// refresh api credentials if needed
+	if !cfg.tokenExpiration.IsZero() && time.Now().Unix() >= cfg.tokenExpiration.Unix() {
```

Comment:
It is better to use `time.Since(cfg.tokenExpiration) > -5*time.Second` condition instead of `time.Now().Unix() >= cfg.tokenExpiration.Unix()`. This should trigger token refresh when less than 5 seconds are left till the expiration.